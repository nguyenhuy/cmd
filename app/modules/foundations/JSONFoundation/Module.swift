Target.module(
  name: "JSONFoundation",
  dependencies: [],
  testDependencies: [])
