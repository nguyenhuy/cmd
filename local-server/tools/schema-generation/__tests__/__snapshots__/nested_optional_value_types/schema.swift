// swiftformat:disable all
// This file is generated from ./local-server/src/server/schemas/schemaFile.ts by `yarn export-schema-swift`.
// Do not edit it manually.

import Foundation
import JSONFoundation

extension Schema {
  public struct ValueType: Codable, Sendable {
    public let array: [String?]?
  
    private enum CodingKeys: String, CodingKey {
      case array = "array"
    }
  
    public init(
        array: [String?]? = nil
    ) {
      self.array = array
    }
  
    public init(from decoder: Decoder) throws {
      let container = try decoder.container(keyedBy: CodingKeys.self)
      array = try container.decodeIfPresent([String?]?.self, forKey: .array)
    }
  
    public func encode(to encoder: Encoder) throws {
      var container = encoder.container(keyedBy: CodingKeys.self)
      try container.encodeIfPresent(array, forKey: .array)
    }
  }}
