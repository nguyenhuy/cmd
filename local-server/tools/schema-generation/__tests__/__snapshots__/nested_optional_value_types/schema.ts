export interface ValueType {
  array?: (string | undefined)[];
}
