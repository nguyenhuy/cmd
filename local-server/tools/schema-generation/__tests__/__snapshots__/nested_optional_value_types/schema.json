{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$ref": "#/definitions/ValueType",
  "definitions": {
    "ValueType": {
      "type": "object",
      "properties": {
        "array": {
          "type": "array",
          "items": {
            "anyOf": [
              {
                "type": "string"
              },
              {
                "not": {}
              }
            ]
          }
        }
      },
      "additionalProperties": false
    }
  }
}